protoc-gen-swagger/genswagger/template.go

```diff
@@ -1189,7 +1196,7 @@ func applyTemplate(p param) (*swaggerObject, error) {
 //
 // If there is no 'Summary', the same behavior will be attempted on 'Title',
 // but only if the last character is not a period.
-func updateSwaggerDataFromComments(swaggerObject interface{}, comment string) error {
+func updateSwaggerDataFromComments(swaggerObject interface{}, comment string, typeName string) error {
```

Comment:
Instead of passing in the typename and adding if cases, can't we skip calling this function altogether if the summary isn't empty?

Reply:
This methods handle specific behavior about setting title/info/description/summary. If we avoid calling this method, this logic needs to be handled on an upper level. The problem is that by default the package title is set to the file name protoc-gen-swagger/genswagger/template.go:949(I don't know why but I suppose we want to keep this behavior) so by changing the behavior of field options taking precedence over comments the title of all protos without package definition was overwritten by the file name so to handle this specific case, the **if** was added

Reply:
I see, thanks. Maybe we can change the parameter from a string to a boolean at least? `IsPackageObject`?

Reply:
sure, I will change it